pulsar-client-cpp/include/pulsar/Hash.h

```diff
@@ -0,0 +1,32 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#ifndef HASH_HPP_
+#define HASH_HPP_
+
+#include <cstdint>
+#include <string>
+
+namespace pulsar {
+class Hash {
```

Comment:
Same for `Hash` , if the user interaction is with the enum, no need to expose this interface. In any case the user won't be able to directly supply a different hash function (other than writing a custom router).